### PR TITLE
Release/1.12.0

### PR DESCRIPTION
**Pull request by thoma202**
Knowledge Base
Fix Order on country which is not enabled
Add information on upgrade that has gone wrong
Load ajax category
Upgrade items specifics loading in ajax
Improve interfae
Add EAN information if wanted

### CODE REVIEW

File: classes/EbaySynchronizer.php [hunk 1]
```diff
@@ -139,23 +140,26 @@ public static function syncProducts($products, $context, $id_lang, $request_cont
...
-				$data['price_percent'] = round( ($price_original - $price) / $price_original * 100.0);
+            if (isset($data['price_original']))
+                $data['price_percent'] = round( ($price_original - $price) / $price_original * 100.0);
 
 			$data['description'] = EbaySynchronizer::_getEbayDescription($product, $ebay_profile, $id_lang);
 
 			// Export to eBay
 			$ebay = EbaySynchronizer::_exportProductToEbay($product, $data, $p['id_ebay_profile'], $ebay_category, $ebay, $date, $context, $id_lang);
+			
+			$handle = fopen(dirname(__FILE__) . '/log-error.txt', 'w');
```
Comment (Quetzacoalt91) [inline classes/EbaySynchronizer.php]: The file opened is not used or closed. You must add the missing lines or remove this one.

File: classes/EbaySynchronizer.php [hunk 2]
```diff
@@ -533,21 +539,29 @@ private static function _hasVariationsMatching($product_id, $id_lang, $ebay_cate
...
 		
-		// convert price to destination currency
-		$currency = new Currency($ebay_profile->getConfiguration('EBAY_CURRENCY'));
-		$price *= $currency->conversion_rate;
-		$price_original *= $currency->conversion_rate;
+		$specific_price_output = null;
+
+		$price = Product::getPriceStatic((int)$product_id, true, null, 6, null, false, true, 1, false, null, null, null, $specific_price_output, true, true, $context);
 		
+		$specific_price_output = null;
```
Comment (Quetzacoalt91) [inline classes/EbaySynchronizer.php]: I do not understand this part. We have a double affectation of the same value in a single variable.
It seems to be better to write `null` directly in the function parameters.

File: classes/EbayTools.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ *  @author    PrestaShop SA <contact@prestashop.com>
+ *  @copyright 2007-2015 PrestaShop SA
+ *  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+class EbayTools
```
Comment (Quetzacoalt91) [inline classes/EbayTools.php]: Why did you copy paste two functions from Tools class ?

**Comment by Quetzacoalt91** [comment]
Hi,

Please fix the issues reported by the validator, you have almost 14000 errors for this PR.
You can fix easily the Standards part by using PHP CS fixer: https://github.com/FriendsOfPHP/PHP-CS-Fixer

**Comment by Quetzacoalt91** [comment]
Take care of the deprecation notices:

> Warning on line 2505 in file /core/www/core_1.6.0.11/classes/Tools.php
[512] Function <b>getDefaultCountryId()</b> is deprecated in <b>/home/pfr81*****/www/ftp/modules/ebay/classes/EbayStat.php</b> on line <b>68</b><br />

**Comment by thoma202** [comment]
Hi @Quetzacoalt91 

I've fixed most of the issue. For the rest, I'm not sure what I need to do to fix them. 
For example : 

image


Concerning the Deprecated method, what do you think is better : 

using something like : 

if(version_compare(_PS_VERSION, '1.5', '<')
        //Then use getDefaultCountryId
else
        // use $this->context->country->id 

?

**Comment by Quetzacoalt91** [comment]
Hi @thoma202,

Does `Configuration::get('PS_COUNTRY_DEFAULT')` help you ?

About your point about the validator error, I cannot currently fix it. It is related to the CodeSniffer package which may require an update.

**Comment by thoma202** [comment]
@Quetzacoalt91 Thanks for the advice :) I've just changed that

**Comment by thoma202** [comment]
Hi @Quetzacoalt91 

Is there any more things you need me to modify ? 
If not do you think you can accept the request on Monday ? 

Thanks a lot.

**Comment by garyjj127** [comment]
Having major problems with the module not listing variations. I've correctly configured eBay categories (and it was previously working),  but it lists variation products separately.

**Comment by thoma202** [comment]
Hi @garyjj127 
Do you mean this specific release, or globally you have an issue with the module ? 
Which category are you trying to synch ? 
Please see this link http://www.202-ecommerce.com/en/partners-modules/ebay-en/ to ask for support

**Comment by thoma202** [comment]
@Quetzacoalt91 
Do you have any news on this ?

**Comment by garyjj127** [comment]
Hi @thoma202. It's globally with the module. It worked up to a couple of weeks ago (1.11), but now it simply won't sync without breaking up the variations. It lists everything separately!

File: classes/EbayKb.php [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+    public function getLink()
+    {
+        if (Tools::isEmpty($this->error_code)) {
+            return false;
+        }
+
+        if ($this->exist()) {
+            $now = new DateTime(date('Y-m-d'));
+            $expire = new DateTime($this->date_upd);
+            $interval = round(($now->format('U') - $now->format('U')) / (60 * 60 * 24));
```
Comment (gaillafr) [inline classes/EbayKb.php]: I think there is a mistake, the following piece of code will always return '0':
```php
($now->format('U') - $now->format('U'))
```

**Comment by thoma202** [comment]
@gaillafr 
I've added licences and fixed the typo :)

**Pull request merged**